docs/framework/lit/quick-start.md

```diff
@@ -0,0 +1,105 @@
+---
+id: quick-start
+title: Quick Start
+---
+
+
+The bare minimum to get started with TanStack Form is to create `FormOptions`and a `TanstackFormController`:
+
+```ts
+import type { FormOptions } from '@tanstack/lit-form'
+
+interface Employee {
+  firstName: string
+  lastName: string
+  employed: boolean
+  jobTitle: string
+}
+
+interface Data {
+  employees: Partial<Employee>[]
+}
+
+const formConfig: FormOptions<Data> = {}
+
+#form = new TanstackFormController(this, formConfig)
+```
+
+In this example `this`references to the instance of your `LitElement` in which you want to use Tanstack Form.
+
+To wire a form element in your template up with Tanstack Form simply use the `field` method of `TanstackFormController`:
+```ts
+render() {
+ return html`
+ <p>Please enter your first name></p>
+ ${this.form.field(
+          {
+            name: `firstName`,
+            validators: {
+              onChange: ({ value }) =>
+                value.length < 3 ? 'Not long enough' : undefined,
+            },
+          },
+          (field: FieldApi<Employee, 'firstName'>) => {
+            return html` <div>
+              <label>First Name</label>
+              <input
+                id="firstName"
+                type="text"
+                placeholder="First Name"
+               @blur="${() => field.handleBlur()}"
+               .value="${field.getValue()}"
+               @input="${(event: InputEvent) => {
+                   if (event.currentTarget) {
+                     const newValue = (event.currentTarget as HTMLInputElement).value;
+                     field.handleChange(newValue);
+                   }
+               }}"
+              />
+            </div>`
+          },
+        )}`;
+}
+```
+The first parameter of `field` is `FieldOptions` and the second is callback to render your element. Be aware that you need
+to handle updating the element and form yourself as seen in the example above. If you do not want to do this, you can use
+the `bind` directive.
+
+## The `bind` directive
+Tanstack Form provides a `bind` directive which makes wiring up elements to the form a bit easier. You no longer need to
+listen to events, update form state or component state, the `bind` directive will do it for you:
+```ts
+render() {
+ return html`
+ <p>Please enter your first name></p>
+ ${this.form.field(
+          {
+            name: `firstName`,
+            validators: {
+              onChange: ({ value }) =>
+                value.length < 3 ? 'Not long enough' : undefined,
+            },
+          },
+          (field: FieldApi<Employee, 'firstName'>) => {
+            return html` <div>
+              <label>First Name</label>
+              <input
+                id="firstName"
+                type="text"
+                placeholder="First Name"
+                ${bind(field)}
```

Comment:
FWIW (and I'd love to hear @fulopkovacs and @aadito123's opinion on this) I don't actually like this `bind` API very much.

I know this came up from convos with the Lit team, but I think our verbosity is the right move. If someone wants to write their own `bind` then that's fine (and we should document how to do so, copy+paste style), but we should not ship our own first-class adapter for any framework in this style IMO

Reply:
I have to agree here. There's a bit of a conflict between the TanStack way and the Lit way. For the sake of consistency across the other form adapters, we probably should not ship bind as part of the library.

Reply:
I am not sure I agree here. Isn't the idea of an adapter to adapt to a given ecosystem?

Reply:
I'm not familiar with the Lit ecosystem and its conventions, but from what I get, all the other adapters aim to be thin wrappers around the core form lib and do not have API-s similar to `bind`. 

<details>
<summary>Handling the field's events in other adapters</summary>

- [React](https://github.com/TanStack/form/blob/eb4801c5202cbe3007401047f3c776351b0631b4/docs/framework/react/quick-start.md?plain=1#L35-L45)
```tsx
            <form.Field
              name="fullName"
              children={(field) => (
                <input
                  name={field.name}
                  value={field.state.value}
                  onBlur={field.handleBlur}
                  onChange={(e) => field.handleChange(e.target.value)}
                />
              )}
            />
```

- [Vue](https://github.com/TanStack/form/blob/eb4801c5202cbe3007401047f3c776351b0631b4/docs/framework/vue/quick-start.md?plain=1#L35-L44)
```vue
      <form.Field name="fullName">
        <template v-slot="{ field }">
          <input
            :name="field.name"
            :value="field.state.value"
            :onBlur="field.handleBlur"
            :onChange="(e) => field.handleChange(e.target.value)"
          />
        </template>
      </form.Field>
```

- [Solid](https://github.com/TanStack/form/blob/eb4801c5202cbe3007401047f3c776351b0631b4/docs/framework/solid/quick-start.md?plain=1#L34-L44)
```tsx
            <form.Field
              name="fullName"
              children={(field) => (
                <input
                  name={field().name}
                  value={field().state.value}
                  onBlur={field().handleBlur}
                  onInput={(e) => field().handleChange(e.target.value)}
                />
              )}
            />
```
</details>

Personally, I like the idea of having utilities like the `bind` directive to reduce the level of verbosity as a user, but they scare me a bit as a maintainer. 😆 The "thin wrapper"-philosophy helps us to not spread ourselves too thin, and allows us to work/iterate faster on the core feature set. I think this is pretty important, especially in the beginning. 

That said, it's also true that the implementation for the `bind` directive does not exactly have the size of a copy-pasteable code snippet: the `packages/lit-form/src/bind.ts` file has 139 lines. Maybe it could be in a separate package, outside of `tanstack/form`? (Or is it too small for that? 🤣)

Reply:
I don't think putting it in another package is really going to be a solution...

Reply:
I think this is the final thing we need to discuss for this PR - we're running into philosophical differences as maintainers (which is okay!).

I think we should avoid shipping `bind`-like APIs in order to avoid maintenance overhead on our end. As soon as we ship one instance of this for one adapter, it opens the potential floodgates for other adapters and then the requests for adapters in every UI library to come in.

I like the approach that @KevinVandy (please feel free to chime in here on this topic!) has done where he maintains different packages externally to TanStack Form for these adapters.

I strongly feel that this is the only way I'd feel comfortable merging this in 😅

Reply:
Resolving per a convo with @Christian24 that I am able to move forward with removing `bind` APIs for now